### ISSUE TITLE
Renderers not working

### ISSUE THREAD

**Issue by ttsanton** [vaadin/vaadin-grid-flow#466]
_Running on 12.0.3_

I have a Grid that gets values with a ListDataProvider.

```
Grid<Apps> grid = new Grid<>();
Set<Apps> appList = getRatings();
ListDataProvider<Apps> listDataProvider = new ListDataProvider<>(appList);

grid.setDataProvider(listDataProvider);
```

I include IconRenderers and TextRenderers for some of the columns for my grid. 

```
grid.addColumn(new IconRenderer<>(
	application -> (application.getCv() != null && !application.getCv().isEmpty())
				? new Icon(VaadinIcon.CHECK_SQUARE)
				: new Icon(VaadinIcon.THIN_SQUARE),
				application -> ""), "cv_url")
    .setHeader("CV").setWidth(wdSmall).setSortable(true);

grid.addColumn(new IconRenderer<>( 
		application -> (application.getPs_url() != null && !application.getPs_url().isEmpty())
				? new Icon(VaadinIcon.CHECK_SQUARE)
				: new Icon(VaadinIcon.THIN_SQUARE),
		application -> ""), "ps_url")
	.setHeader("PS").setWidth(wdSmall);

grid.addColumn(new IconRenderer<>(
		application -> (application.getLor_url() != null && !application.getLor_url().isEmpty())
				? (application.getLor_url().equals("N/A")) ? new Icon(VaadinIcon.RASTER)
						: new Icon(VaadinIcon.CHECK_SQUARE)
				: new Icon(VaadinIcon.THIN_SQUARE)))
	.setHeader("LoR").setWidth(wdSmall).setSortable(true);
```

However, when the page loads, the Renderer does not show up right away:

![screen shot 2018-12-17 at 3 55 48 pm](https://user-images.githubusercontent.com/7467196/50115035-3f882800-0214-11e9-9dac-e2982da45480.png)

If I force the Grid to sort, it will show up, but still with errors:
![screen shot 2018-12-17 at 3 56 38 pm](https://user-images.githubusercontent.com/7467196/50115080-621a4100-0214-11e9-8448-22ea3ba2a076.png)

I tried with using a normal setItems() call for the Grid, but the same results happen.

I also added some print statements to check and see when the Renderers are actually being used, and they are not triggered until after I manually sort the grid or if I scroll down further down a Grid with lots of rows, forcing it to load in more items.

**Comment by denis-anisimov** [comment]
Source code is not complete. I cannot use to reproduce.

The code which I've written based on this snippet works for me as expected.

Please use https://github.com/vaadin/skeleton-starter-flow to make a complete project which I may use to reproduce the issue.

**Comment by ttsanton** [comment]
@denis-anisimov , I have been able to fix the problem I had with the renderers initially not showing up on the Grid.

I don't know how to fully explain in detail, but I am working with a set of data that can hold a variable amount of columns and needs to be updated based on a ComboBox selection. As a result, I needed to use the Renderers as a way to "match" information into the respective rows. 
In Vaadin 7, I was able to achieve this by creating a version of the addRow() method that worked with my data, but this Vaadin version requires me to use the Renderers as a way to associate the correct information to its repective rows.

I thought I could achieve this by loading in the ListDataProvider to the same grid each time and just re-create the columns using this code snippet:
```
//remove all columns and re-build
for(Column<Apps> c : columns)
	grid.removeColumn(c);

grid.addColumn(...)
... 
//re-fill the columns for the grid
```
But, it appears that I need to create a new grid each time, since the number of columns I might have can change.
